### PR TITLE
[Merged by Bors] - feat(init/meta/interative): Change the behaviour of `specialize` to put goals from underscores first

### PR DESCRIPTION
**Pull request by shingtaklam1324**
originally leanprover-community/mathlib#6150

### CODE REVIEW

**Comment by shingtaklam1324** [comment]
There is also a set of tests at https://github.com/leanprover-community/mathlib/pull/6150/files#diff-bf742064f8ef6e229c0fa7b5df4c4b97b090242eae93579926f970c5856ccaa8 but I'm not sure how the tests work in this repo.

**Comment by bryangingechen** [comment]
There's some explanation [here](https://github.com/leanprover-community/lean/blob/master/doc/fixing_tests.md), but it should also be fine to let CI tell us what's broken:

bors try

**Comment by bryangingechen** [comment]
bors try

**Comment by bryangingechen** [comment]
bors ping

File: library/init/meta/interactive.lean [hunk 1]
```diff
@@ -1636,11 +1636,12 @@ do gs ← get_goals,
 /--
 The tactic `specialize h a₁ ... aₙ` works on local hypothesis `h`. The premises of this hypothesis, either universal quantifications or non-dependent implications, are instantiated by concrete terms coming either from arguments `a₁` ... `aₙ`. The tactic adds a new hypothesis with the same name `h := h a₁ ... aₙ` and tries to clear the previous one.
 -/
-meta def specialize (p : parse texpr) : tactic unit :=
+meta def specialize' (p : parse texpr) : tactic unit :=
```
Comment (bryangingechen) [inline library/init/meta/interactive.lean:1639]: Ah, you probably don't want to change the name here!
```suggestion
meta def specialize (p : parse texpr) : tactic unit :=
```

Comment (shingtaklam1324) [reply library/init/meta/interactive.lean]: ah yes, I've copied this from the PR. The prime shouldn't be there.

**Comment by bryangingechen** [comment]
bors try

**Pull request closed without merging**